app/components/primer/navigation/tab_component.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+module Primer
+  module Navigation
+    # This component is part of navigation components such as `Primer::TabNavComponent`
+    # and `Primer::UnderlineNavComponent` and should not be used as a standalone component.
+    class TabComponent < Primer::Component
+      include ViewComponent::SlotableV2
+
+      # Panel controlled by the Tab. It will only renderd if `with_panel` is `true`.
+      #
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      renders_one :panel, lambda { |**system_arguments|
+        system_arguments[:tag] ||= :div
+        system_arguments[:role] ||= :tabpanel
+        system_arguments[:hidden] = true unless @selected
+
+        Primer::BaseComponent.new(**system_arguments)
+      }
+
+      # Icon to be rendered in the Tab left.
+      #
+      # @param kwargs [Hash] The same arguments as <%= link_to_component(Primer::OcticonComponent) %>.
+      renders_one :icon, lambda { |**system_arguments|
+        system_arguments[:classes] = class_names(
+          @icon_classes,
+          system_arguments[:classes]
+        )
+        Primer::OcticonComponent.new(**system_arguments)
+      }
+
+      # The Tab's title.
+      #
+      # @param kwargs [Hash] The same arguments as <%= link_to_component(Primer::TextComponent) %>.
+      renders_one :title, Primer::TextComponent
+
+      # Counter to be rendered in the Tab right.
+      #
+      # @param kwargs [Hash] The same arguments as <%= link_to_component(Primer::CounterComponent) %>.
+      renders_one :counter, Primer::CounterComponent
+
+      attr_reader :selected
+
+      # @example Default
+      #   <%= render(Primer::Navigation::TabComponent.new(selected: true)) do |c| %>
+      #     <% c.title { "Selected" } %>
+      #   <% end %>
+      #   <%= render(Primer::Navigation::TabComponent.new) do |c| %>
+      #     <% c.title { "Not selected" } %>
+      #   <% end %>
+      #
+      # @example With icons and counters
+      #   <%= render(Primer::Navigation::TabComponent.new) do |c| %>
+      #     <% c.icon(icon: :star) %>
```

Comment:
Gosh, I really wish `OcticonComponent` took positional arguments. This would be cleaner as:

```suggestion
      #     <% c.icon(:star) %>
```

Reply:
we could potentially change
```rb
    def initialize(icon:, size: SIZE_DEFAULT, **system_arguments)
      @icon = icon
```

to

```rb
    def initialize(icon_name = nil, icon: nil, size: SIZE_DEFAULT, **system_arguments)
      @icon = icon_name || icon
```

which would avoid breaking changes?

Reply:
I like that. Could you file an issue?